Content.Server/Pinpointer/NavMapSystem.cs

```diff
@@ -0,0 +1,106 @@
+using Content.Shared.Pinpointer;
+using Content.Shared.Tag;
+using Robust.Shared.GameStates;
+using Robust.Shared.Map.Components;
+using Robust.Shared.Physics;
+using Robust.Shared.Physics.Components;
+using Robust.Shared.Timing;
+
+namespace Content.Server.Pinpointer;
+
+/// <summary>
+/// Handles data to be used for in-grid map displays.
+/// </summary>
+public sealed class NavMapSystem : SharedNavMapSystem
+{
+    [Dependency] private readonly IGameTiming _timing = default!;
+    [Dependency] private readonly TagSystem _tags = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<AnchorStateChangedEvent>(OnAnchorChange);
+        SubscribeLocalEvent<GridInitializeEvent>(OnGridInit);
+        SubscribeLocalEvent<NavMapComponent, ComponentGetState>(OnGetState);
+    }
+
+    private void OnGridInit(GridInitializeEvent ev)
+    {
+        EnsureComp<NavMapComponent>(ev.EntityUid);
```

Comment:
Right now station maps works on any grid (including shuttles, salvage, etc). Do you really need this ensure? I think mappers can do that manually.

Reply:
It's kind of a hassle to track everything for mapping, there's already half a dozen components that are added via code otherwise the map may just break.